lib/core/connect.js

```diff
@@ -79,15 +79,17 @@ class Connector {
     socket
       .setNoDelay(true)
       .once(protocol === 'https:' ? 'secureConnect' : 'connect', function () {
-        if (callback) {
-          clearTimeout(timeout)
+        clearTimeout(timeout)
 
+        if (callback) {
           const cb = callback
           callback = null
           cb(null, this)
         }
       })
       .on('error', function (err) {
+        clearTimeout(timeout)
+
         if (callback) {
```

Comment:
Probably enough to just put a `clearTimeout(timeout)` here.

Reply:
Why?

Reply:
The problem is that we don't clear it in the error case. In the non error case it is already cleared.

Reply:
It's fine the way you did it as well.